utils/s2n_array.h

```diff
@@ -29,11 +34,16 @@ struct s2n_array {
 
     /* The size of each element in the array */
     uint32_t element_size;
+
+    /* The growable status of the array */
+    s2n_array_growable growable;
 };
 
 extern S2N_RESULT s2n_array_validate(const struct s2n_array *array);
 extern struct s2n_array *s2n_array_new(uint32_t element_size);
+extern struct s2n_array *s2n_array_new_with_capacity(uint32_t element_size, uint32_t capacity);
 extern S2N_RESULT s2n_array_init(struct s2n_array *array, uint32_t element_size);
+extern S2N_RESULT s2n_array_set_growable(struct s2n_array *array, s2n_array_growable growable);
 extern S2N_RESULT s2n_array_pushback(struct s2n_array *array, void **element);
```

Comment:
This will prevent the array from resizing, but there's one other expensive behavior for this data structure: if you insert at an existing index, all the later entries have to be copied. That's also unnecessary and expensive for your use case.

Could you have some API that disables all the expensive features of an s2n_array? Maybe "s2n_array_set_static", "s2n_array_lock", or "s2n_array_disable_dynamic_features"?

Otherwise we have to rely on the fact that you're not currently calling s2n_array_insert. But one of my issues with this data structure is that it does expensive operations like that copy basically silently, so if someone later naively swaps your pushback out for inserting at the beginning of the list, they won't necessarily know that's a mistake and they're now triggering a copy on every insert. Performance will just slightly degrade.

Reply:
That makes sense! I changed `set_growable` to `set_static`, which now also disables shifting. I added new tests for this as well.